packages/rs-dpp/src/document/state_transition/documents_batch_transition/mod.rs

```diff
@@ -57,15 +72,74 @@ impl std::default::Default for DocumentsBatchTransition {
             transition_type: StateTransitionType::DocumentsBatch,
             owner_id: Identifier::default(),
             transitions: vec![],
-            signature_public_key_id: 0,
-            signature: vec![],
+            signature_public_key_id: None,
+            signature: None,
             execution_context: Default::default(),
         }
     }
 }
 
 impl DocumentsBatchTransition {
-    // TODO (rs-dpp-feature): do not use [`JsonValue`] with constructors
+    pub fn from_json_object(
+        json_value: JsonValue,
+        data_contracts: Vec<DataContract>,
+    ) -> Result<Self, ProtocolError> {
+        let mut json_value = json_value;
+
+        let maybe_signature = json_value.get_string(property_names::SIGNATURE).ok();
+        let signature = if let Some(signature) = maybe_signature {
+            Some(base64::decode(signature).context("signature exists but isn't valid base64")?)
+        } else {
+            None
+        };
+
+        let mut batch_transitions = DocumentsBatchTransition {
+            protocol_version: json_value
+                .get_u64(property_names::PROTOCOL_VERSION)
+                // js-dpp allows `protocolVersion` to be undefined
+                .unwrap_or(LATEST_VERSION as u64) as u32,
+            signature,
+            signature_public_key_id: json_value
+                .get_u64(property_names::SIGNATURE_PUBLIC_KEY_ID)
+                .ok(),
+            owner_id: Identifier::from_string(
+                json_value.get_string(property_names::OWNER_ID)?,
+                Encoding::Base58,
+            )?,
+            ..Default::default()
+        };
+
+        let mut document_transitions: Vec<DocumentTransition> = vec![];
+        let maybe_transitions = json_value.remove(property_names::TRANSITIONS);
```

Comment:
I'm not sure why `remove`?

Reply:
To avoid clonig. Later we manually re-create every transition using `DocumentTransition::from_json_object()` which moves whole value.